### ISSUE TITLE
Xpressbazaar - LSP - compliance check 

### ISSUE THREAD

**Issue by abhinavv245** [ONDC-Official/v1.2.0-logs#1974]
## Flow 1

**/on_search**

- bpp_terms (static terms) in bpp/descriptor are not enabled yet, need not be sent
- For TAT 4H, child category of Standard Delivery i.e. Same Day Delivery can be used
- what logic has been used to calculate motorable distance?
- providers/locations to be provided in case LSP wants the logistics buyer to drop off the package at LSP location

**/on_init**

- cancellation terms are not enabled yet, also, cancellation fee, if any, for this state & reason code - either percentage (of order value minus taxes, in the above format between "0.00" and "100.00") or amount will be provided later (in cancellation terms)

**/confirm**

- why is pickup and delivery time (/fulfillments/start and end/time) range and timestamp provided in confirm call by BAP?

**/on_confirm**

- why is S2D TAT in items/time different than /confirm?
- order/updated_at should be updated w.r.t context/timestamp
- Average pickup time is 20 min, why is the pickup range provided for the next day? Correlate average pickup time, pickup range, TAT and delivery range.
- tracking should be enabled (true) for hyperlocal P2P orders

**/on_status**

-  tracking attributes in fulfillments/tags to be provided, on or before fulfillment state of "Agent-assigned"
- Pickup timestamp (fulfillments/start/time/timestamp) cannot be greater than or equal to  delivery timestamp (fulfillments/end/time/timestamp) for fulfillment state - Order-delivered"

**track/on_track to be implemented**

## Flow 2 (RTO)

**/on_cancel**

- unsolicited on_cancel to be implemented due to buyer unavailability 
- Message ID should be same as /cancel: \"msgmp2i_o4phamdpjemk_urcoy3
- order must NOT have additional properties (breakup)  (should be inside quote object)
- cancellation/reason/id does not match the cancellation reason id in /cancel
- fulfillment_id is an invalid attribute in breakup
- tax line item is missing for delivery and RTO charges
- quote must have required property 'price'
- fulfillments/0/id does not match with the one in /on_confirm
- fulfillments/0/type must be equal to one of the allowed values (Delivery,Return,RTO)
fulfillments/1/end/location must have required property 'gps'
- fulfillments/1/end/location must have required property 'address'
- billing/created_at mismatches in /billing from /confirm"
- billing/updated_at mismatches in /billing from /confirm"
- cancellation/cancelled_by should be the subscriber id of the entity who cancelled the order
- fulfillments/1 (RTO fulfillment) - end/location/location is invalid, time/duration and instructions are not required, end/timestamp should not be there as RTO is not delivered yet, time range should be there
- precancel state, rto event tags are required for the cancelled fulfillment

**/on_status**

- order must have required property 'cancellation' after the order is cancelled 
- RTO fulfillment with the updated state is missing (complete order trail to be maintained)

## Flow 3

**/on_cancel**

- rto event tag is not required as the order has not been cancelled by LSP
- how is precancel state 'Order-picked-up'? (should be Pending), updated_at should also be recorded correctly
- quote should reflect the updated value of logistics order (in case no cancellation fee is being charged before agent is assigned or pickup, should be updated to zero)

@deathevil0707

**Comment by pzed05** [comment]
@abhinavv245 , logs as per the feedback for Lsp Seller App all flows (Xpressbaazaar), please verify the logs
https://github.com/ONDC-Official/v1.2.0-logs/pull/2145

**Comment by abhinavv245** [comment]
**/on_search**

- bpp_terms (static terms) in bpp/descriptor are not enabled yet, tags need not be sent
- what logic has been used to calculate motorable distance? (distance shows to be around 24km)
- Average pickup time(fulfillments/start/time/duration) should be according to the S2D TAT, for standard delivery TAT of 1 day, how can average pickup time be also 1 day?

**/on_init**

- as payment is collected by BAP, settlment details will be sent by seller app

**/on_status**

- provider/locations/id is invalid here

## RTO flow

**/on_status**

- why are multiple on_status calls sent for RTO-Disposed?

@pzed05

**Comment by yogita888** [comment]
@abhinavv245 , logs as per feedback , please review this pr
https://github.com/ONDC-Official/v1.2.0-logs/pull/2190

**Comment by abhinavv245** [comment]
**/on_search**

- items/time/timestamp should be as per the TAT provided w.r.t context/timestamp (for RTO item as well)

**/on_status**

- fulfillments/start/duration mismatch from /on_search

## RTO Flow

**/on_cancel (RTO)**

- precancel_state/updated_at timestamp is not recorded correctly for 'Out-for-delivery' state
- settlement details should be of the seller app, also dummy data to be removed (in all APIs)

**/on_status (RTO-Delivered)**

- cancellation_reason_id is incorrect in fulfillments/tags (rto_event) and cancellation object

## Deferred RTS

**/on_update**

- ready_to_ship tag should 'yes'

@yogita888

**Comment by Somnathens** [comment]
@abhinavv245 , logs as per your feedback, Please review this pr
https://github.com/ONDC-Official/v1.2.0-logs/pull/2222

**Comment by BLR-0118** [comment]
Xpressbazaar logs show the following:
bap_id - preprod.xpressbaazaar.com;
bpp_id - onpp.grouptci.com;
who's this bpp?
@Somnathens @yogita888

**Comment by Somnathens** [comment]
@BLR-0118 sir, bpp_id: onpp.gouptci.com is for Transport Corporation of India (TCI).

**Comment by BLR-0118** [comment]
is this thru ENS/Xpressbazaar?

**Comment by Somnathens** [comment]
@BLR-0118 , Yes it's through ENS as a TSP for both b2c and b2b logistics seller.

**Comment by BLR-0118** [comment]
in that case, you need to submit logs with different LBNP

**Comment by Somnathens** [comment]
@BLR-0118 sir, our logs has been approved for logistics buyer app having link:
https://github.com/ONDC-Official/v1.2.0-logs/issues/1485
So, Testing has been done with verified logistics buyer app. 
If its mandatory to test with different buyer then please align a logistics buyer with whom we can do this testing part (We support standard delivery with P2P type delivery).

**Comment by Somnathens** [comment]
@abhinavv245 , @BLR-0118 , we have tested with different LBNP,
Please review this pr and take appropriate action 
https://github.com/ONDC-Official/v1.2.0-logs/pull/2302

**Comment by BLR-0118** [comment]
@abhinavv245 - pls check this

**Comment by abhinavv245** [comment]
## ORDER FLOW

- implement tracking

**/on_init**

- settlment details should be of the LSP as payment is collected by BAP

**/on_status**

- What does tracking enabled for both url and gps ('yes') signifies here? (fulfillments/tags (tracking))

## CANCEL FLOW

- **/on_track**

- how is tracking active before agent is assigned?

@Somnathens

**Comment by Somnathens** [comment]
@BLR-0118 , @abhinavv245 we have submitted logs as per  given above  feedback.
please review this pull request: https://github.com/ONDC-Official/v1.2.0-logs/pull/2403
 Looking forward to your response and please give response as earlier possible.
